Sources/ScipioKit/FileSystem.swift

```diff
@@ -0,0 +1,209 @@
+    /// - Throws: An error if the write operation fails.
+    func writeFileContents(_ path: URL, string: String) throws {
+        guard let data = string.data(using: .utf8) else {
+            throw FileSystemError.utf8EncodingFailed
+        }
+        try writeFileContents(path, data: data)
+    }
+
+    /// Checks if the item at the specified URL exists, following symbolic links.
+    /// - Parameter path: The URL to check.
+    /// - Returns: True if the path exists; false otherwise.
+    func exists(_ path: URL) -> Bool {
+        exists(path, followSymlink: true)
+    }
+}
+
+public struct LocalFileSystem: FileSystem {
+    public static let `default` = LocalFileSystem()
+
+    public var tempDirectory: URL {
+        FileManager.default.temporaryDirectory
+    }
+
+    public var cachesDirectory: URL? {
+        FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first
+    }
+
+    public var currentWorkingDirectory: URL? {
+        URL(filePath: FileManager.default.currentDirectoryPath)
+    }
+
+    public func writeFileContents(_ path: URL, data: Data) throws {
+        guard path.isFileURL else {
+            throw FileSystemError.invalidFileURL(path: path)
+        }
+
+        if !exists(path) {
+            try createDirectory(
+                path.deletingLastPathComponent(),
+                recursive: true
+            )
+        }
+
+        try data.write(to: path, options: .atomic)
+    }
+
+    public func readFileContents(_ path: URL) throws -> Data {
+        guard let contents = FileManager.default.contents(atPath: path.path(percentEncoded: false)) else {
+            throw FileSystemError.cannotReadFileContents(path: path)
+        }
+        return contents
+    }
+
+    public func exists(_ path: URL, followSymlink: Bool = true) -> Bool {
+        if followSymlink {
+            return FileManager.default.fileExists(atPath: path.path(percentEncoded: false))
+        }
+        return (try? FileManager.default.attributesOfItem(atPath: path.path(percentEncoded: false))) != nil
+    }
+
+    public func isDirectory(_ path: URL) -> Bool {
+        var isDirectory: ObjCBool = false
+        let exists = FileManager.default.fileExists(atPath: path.path(percentEncoded: false), isDirectory: &isDirectory)
+        return exists && isDirectory.boolValue
+    }
+
+    public func isFile(_ path: URL) -> Bool {
+        let path = path.resolvingSymlinksInPath()
+        let attributes = try? FileManager.default.attributesOfItem(atPath: path.path(percentEncoded: false))
+        return attributes?[.type] as? FileAttributeType == .typeRegular
+    }
+
+    public func isSymlink(_ path: URL) -> Bool {
+        let values = try? path.resourceValues(forKeys: [.isSymbolicLinkKey])
+        return values?.isSymbolicLink == true
+    }
+
+    public func copy(from fromURL: URL, to toURL: URL) throws {
+        try FileManager.default.copyItem(at: fromURL, to: toURL)
+    }
```

Comment:
ref: https://github.com/swiftlang/swift-tools-support-core/blob/f9b401016b70c6b8409e5c97e74d97513d1a8d02/Sources/TSCBasic/FileSystem.swift#L623-L628

The original source existence check and destination non-existence check are not required?